src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4336,21 +4328,57 @@ private void VisitDeconstructionArguments(ArrayBuilder<DeconstructionVariable> v
                 for (int i = 0; i < n; i++)
                 {
                     var variable = variables[i];
+                    var underlyingConversion = conversion.UnderlyingConversions[i];
                     var rightPart = rightParts[i];
                     var nestedVariables = variable.NestedVariables;
                     if (nestedVariables != null)
                     {
-                        VisitDeconstructionArguments(nestedVariables, conversion.UnderlyingConversions[i], rightPart);
+                        VisitDeconstructionArguments(nestedVariables, underlyingConversion, rightPart);
                     }
                     else
                     {
                         var targetType = variable.Type;
-                        var valueType = VisitOptionalImplicitConversion(rightPart, targetType, useLegacyWarnings: true, AssignmentKind.Assignment);
-                        int valueSlot = MakeSlot(rightPart);
-                        int slotWatermark = GetSlotWatermark();
-                        // targetSlot is calculated last to avoid copying from that slot in cycles.
+                        TypeSymbolWithAnnotations operandType;
+                        TypeSymbolWithAnnotations valueType;
+                        int valueSlot;
+                        if (underlyingConversion.IsIdentity)
```

Comment:
>if (underlyingConversion.IsIdentity) [](start = 24, length = 36)

It looks like there is a lot of specialized code here. Can we simply do what ``VisitAssignmentOperator``` does? Perhaps share code with it? It feels like this operation should be very similar to a regular assignment case.

Reply:
Since `underlyingConversions` are not included as `BoundNodes`, it was not clear how to share code cleanly. Added a comment.

---
In reply to: [258285342](https://github.com/dotnet/roslyn/pull/33455#discussion_r258285342) [](ancestors = 258285342)